bpf/bpf_host.c

```diff
@@ -1361,6 +1361,9 @@ int cil_to_netdev(struct __ctx_buff *ctx __maybe_unused)
 #endif
 
 #ifdef ENABLE_HOST_FIREWALL
+	if (ctx_snat_done(ctx))
+		goto skip_host_firewall;
+
```

Comment:
The reasoning makes sense, although the scope of the change is a bit wider (it also affects eg. NAT-forwarded LB connections in TC, and RevDNATed replies, ...).

Using the SNAT-done flag for such policy decisions feels a bit fragile. I would hope that we can come up with a better solution over time. But good enough for me atm :+1:.